sw/device/tests/plic_all_irqs_test.h

```diff
@@ -0,0 +1,85 @@
+    bool is_pending;                                                          \
+    CHECK(dif_##peripheral##_irq_is_pending(&handle, irq, &is_pending) ==     \
+              kDifOk,                                                         \
+          "dif_" #peripheral "_irq_is_pending failed");                       \
+    CHECK(is_pending, "Interrupt %d fired at PLIC did not fire at " #handle,  \
+          irq);                                                               \
+                                                                              \
+    /* Clear the interrupt at the peripheral. */                              \
+    CHECK(dif_##peripheral##_irq_acknowledge(&handle, irq) == kDifOk,         \
+          "dif_" #peripheral "_irq_acknowledge failed");                      \
+  } while (0)
+
+/**
+ * Initializes the handle to a peripheral instance.
+ */
+#define PERIPHERAL_INIT(peripheral, handle, base_addr)      \
+  do {                                                      \
+    mmio_region_t addr = mmio_region_from_addr(base_addr);  \
+    CHECK(dif_##peripheral##_init(addr, &handle) == kDifOk, \
+          "dif_" #peripheral "_init failed");               \
+  } while (0)
+
+/**
+ * Enables all IRQs in the given peripheral instance.
+ */
+#define PERIPHERAL_IRQS_ENABLE(peripheral, handle)                          \
+  do {                                                                      \
+    dif_##peripheral##_irq_state_snapshot_t all_irqs =                      \
+        (dif_##peripheral##_irq_state_snapshot_t)ULONG_MAX;                 \
+    CHECK(dif_##peripheral##_irq_restore_all(&handle, &all_irqs) == kDifOk, \
+          "dif_" #peripheral "_irq_restore_all failed");                    \
+  } while (0)
+
+/**
+ * Triggers all IRQs in a peripherals instance one-by-one.
+/*/
+#define PERIPHERAL_IRQS_TRIGGER(peripheral, handle, plic_peripheral,           \
+                                start_irq, end_irq)                            \
+  do {                                                                         \
+    peripheral_expected = plic_peripheral;                                     \
+    for (dif_##peripheral##_irq_t irq = start_irq; irq <= end_irq; ++irq) {    \
+      peripheral##_irq_expected = irq;                                         \
+      LOG_INFO("Triggering " #handle " IRQ %d.", irq);                         \
+      CHECK(dif_##peripheral##_irq_force(&handle, irq) == kDifOk,              \
+            "dif_" #peripheral "_irq_force failed");                           \
+      /* NOTE; The interrupt triggered above is instantaneous - Ibex jumps     \
+       * into the ISR even before a WFI can be called. Hence, we add a couple  \
+       * of nops instead. */                                                   \
+      asm volatile("nop");                                                     \
+      asm volatile("nop");                                                     \
```

Comment:
Hang on, why do you need these no-ops? Presumably Ibex enters the handler before irq_force even returns, so it's not clear what effect these are supposed to have.

Reply:
Ahh yes, I was experimenting with these. They are not needed. Removed.